### PR TITLE
Change default to not throw exception for a call on a zeroed out assay

### PR DESCRIPTION
**Pull request by gbggrant**
Fix bug in unsignedShortToInt (and add tests)

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/arrays/GtcToVcf.java [hunk 1]
```diff
@@ -149,6 +149,9 @@
     @Argument(shortName = "FP_VCF", doc = "The fingerprint VCF for this sample", optional = true)
     public File FINGERPRINT_GENOTYPES_VCF_FILE;
 
+    @Argument(doc = "", optional = true)
```
Comment (yfarjoun) [inline src/main/java/picard/arrays/GtcToVcf.java]: documentation

Comment (gbggrant) [reply src/main/java/picard/arrays/GtcToVcf.java]: oops....

File: src/main/java/picard/arrays/GtcToVcf.java [hunk 2]
```diff
@@ -383,7 +385,12 @@ private VariantContext makeVariantContext(Build37ExtendedIlluminaManifestRecord
         if (egtFile.totalScore[egtIndex] == 0.0) {
             builder.filter(InfiniumVcfFields.ZEROED_OUT_ASSAY);
             if (genotype.isCalled()) {
-                throw new PicardException("Found a call on a zeroed out SNP!!");
+                if (DO_NOT_ALLOW_CALLS_ON_ZEROED_OUT_ASSAYS) {
+                    throw new PicardException("Found a call on a zeroed out Assay!!");
```
Comment (pshapiro4broad) [inline src/main/java/picard/arrays/GtcToVcf.java]: Would it be helpful to dump `genotype` here too?

File: src/test/java/picard/arrays/illumina/InfiniumDataFileTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    public void testFloatToByteArray(byte[] expectedBytes, float value) {
+        byte[] themBytes = InfiniumDataFile.floatToByteArray(value);
+        Assert.assertEquals(themBytes, expectedBytes);
+    }
+
+    @Test(enabled = false)
+    public void testByteArrayToCharArray() {
+        byte[] themBytes = new byte[]{77, 69, 71, 95};
+        char[] foo = InfiniumDataFile.byteArrayToCharArray(themBytes);
+        Assert.assertEquals(String.valueOf(foo), "MEG_");
```
Comment (pshapiro4broad) [inline src/test/java/picard/arrays/illumina/InfiniumDataFileTest.java]: Does this work? I see that the test is disabled. Maybe you want an `assertEquals()` on two `char` arrays?

File: src/main/java/picard/arrays/GtcToVcf.java (hunk 1 above)
Comment (gbggrant) [reply src/main/java/picard/arrays/GtcToVcf.java]: added

File: src/main/java/picard/arrays/GtcToVcf.java (hunk 2 above)
Comment (gbggrant) [reply src/main/java/picard/arrays/GtcToVcf.java]: sure.

File: src/test/java/picard/arrays/illumina/InfiniumDataFileTest.java (hunk 3 above)
Comment (gbggrant) [reply src/test/java/picard/arrays/illumina/InfiniumDataFileTest.java]: Yes, this works.  Tests were disabled during testing - now readded.

**Pull request merged**